AdventureWorks.Core.Interfaces/Persistance/IGenericRepository.cs

```diff
@@ -1,23 +1,22 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq.Expressions;
 
 namespace AdventureWorks.Core.Interfaces.Persistance
 {
-    public interface IRepository<TEntity> where TEntity : class
+    public interface IGenericRepository<TEntity> where TEntity : class
     {
-        IEnumerable<TEntity> GetAll();
+        IEnumerable<TEntity> GetAll(Func<TEntity, bool> predicate = null);
 
-        IEnumerable<TEntity> FindBy(Expression<Func<TEntity, bool>> predicate);
+        TEntity GetById(string id);
```

Comment:
Add common methods instead of add the same method in every particular repository